open-apparel-registry/open-apparel-registry
# Cache vector tiles for a year

## Overview

Add a `Cache-Control: max-age=31536000` header to the vector tile responses. This allows the browser to cache the tile for a day.

This header will also be used by Cloudfront to cache the tiles after the first user request is made. If we'd like Cloudfront to have a separate cache duration than browsers, we can add `, s-maxage=CLOUDFRONT_CACHE_DURATION` to the end of the `Cache-Control` header. See [here for more details](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Expiration.html#ExpirationDownloadDist).

The cache duration is currently set to be a day, but this can be changed on feedback. This can also be made into an environment variable if we'd like to be able to change this at run time.

Also add some configuration for Cloudfront that ensures it caches the tiles generated by a user request.

Request review from both app and ops teams since this touches both aspects. Need eyes on the Terraform configuration, which was updated [according to this](https://www.terraform.io/docs/providers/aws/r/cloudfront_distribution.html#ordered_cache_behavior).

~Did not connect to #763 or #764 since this does not include S3 work.~

Connects #779 

## Demo

![image](https://user-images.githubusercontent.com/1430060/64452369-40b7f880-d0b4-11e9-8141-63a41accbbeb.png)

## Testing Instructions

* Check out this branch and ensure that the vector tile flag is turned on `manage waffle_switch vector_tile on`
* Run `server` and go to [:6543/](http://localhost:6543/)
* Open the developer tools and ensure that "Disable Cache" is unchecked
* Observe the vector tile responses. Wait for them to finish.
* Refresh the page. Ensure the browser uses cached responses the second time.
* Update a facility to change the cache key. Or artificially change it.
* Refresh the page. Ensure the browser fetches fresh tiles.

## Checklist
